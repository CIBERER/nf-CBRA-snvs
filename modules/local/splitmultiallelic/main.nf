process SPLITMULTIALLELIC {
    tag "${meta.id}_${meta.program}"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bcftools:1.20--h8b25389_0':
        'biocontainers/bcftools:1.20--h8b25389_0' }"

    input:
    tuple val(meta), path(vcf), path(tbi)
    tuple val(meta2), path(fasta)
    val(program)

    output:
    tuple val(meta), path("${prefix}.biallelic.${program}.vcf.gz"), path("${prefix}.biallelic.${program}.vcf.gz.tbi"), emit: biallelic_renamed_vcf
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    //program = task.ext.prefix ?: "${meta.program}"

    """
    bcftools query -l ${vcf} > samples.${program}.txt
    sed -e 's/\$/.${program}/' -i samples.${program}.txt

    bcftools reheader --samples samples.${program}.txt -o ${prefix}.renamed.${program}.vcf.gz ${vcf}
    tabix -p vcf ${prefix}.renamed.${program}.vcf.gz

    bcftools norm -m - -c s -f ${fasta} ${prefix}.renamed.${program}.vcf.gz | bcftools view --min-ac=1 -O z -o ${prefix}.biallelic.${program}.vcf.gz 
    tabix -p vcf ${prefix}.biallelic.${program}.vcf.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bcftools: \$(bcftools --version 2>&1 | head -n1 | sed 's/^.*bcftools //; s/ .*\$//')
    END_VERSIONS
    """
}