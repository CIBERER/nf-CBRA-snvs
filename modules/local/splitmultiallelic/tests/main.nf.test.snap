{
    "splitmultiallelic - vcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.biallelic.gatk.vcf.gz:md5,01e9b8d3417e25d284da9d3d9aac4d40",
                        "test_sample.biallelic.gatk.vcf.gz.tbi:md5,5a54d4b522943df5bc06734a0b67d6fd"
                    ]
                ],
                "1": [
                    "versions.yml:md5,9fdfc65b78f7ac5e2d0ed7cd0209357a"
                ],
                "biallelic_renamed_vcf": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.biallelic.gatk.vcf.gz:md5,01e9b8d3417e25d284da9d3d9aac4d40",
                        "test_sample.biallelic.gatk.vcf.gz.tbi:md5,5a54d4b522943df5bc06734a0b67d6fd"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9fdfc65b78f7ac5e2d0ed7cd0209357a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-05-07T08:11:31.861527024"
    }
}